script/node_bs.py

```diff
@@ -0,0 +1,281 @@
+from distutils.log import ERROR, INFO, WARN
+from logging import WARNING
+
+import os
+import json
+import requests
+import subprocess
+from subprocess import call
+import glob
+import stat
+import docker
+import shutil
+import sys
+from datetime import date
+
+
+GITHUB_PACKAGE_INFO_API = "https://api.github.com/repos/{}/{}/{}/{}"
+
+
+path_separator = os.path.sep
+#ROOT = os.path.dirname(os.path.dirname(__file__))
+ROOT = os.getcwd()
+package_name = input("Enter Package name (Package name should match with the directory name): ")
+#package_name = 'elasticsearch'
+package_name = package_name.lower()
+dir_name = f"{ROOT}{path_separator}{package_name[0]}{path_separator}{package_name}"
+
+
+github_url=''
+latest_release=''
+active_repo=False
+new_build_script=''
+branch_pkg=""
+
+
+def get_latest_build_script(dir_name):
+    files_path = os.path.join(dir_name,"*")
+    files = sorted(glob.iglob(files_path),key=os.path.getctime,reverse=True)
+    latest_existing_build_script =""
+    for script_file in files:
+        if script_file.endswith(".sh") and (script_file.find("ubuntu")==-1):
+            global github_url
+            github_url = get_repo_url(script_file)
+            global active_repo
+            if github_url!="NUrl" and github_url!='':
+                if (check_repo_activeness(github_url)):
+                    active_repo = True
+                    global latest_release
+                    latest_release = get_latest_release(github_url)
+                else:
+                    active_repo = False
+                    print("\n Repo is Not active . Stopping the further process.")
+                    exit()
+
+            return script_file
+
+    return "NPresent"
+
+def get_repo_url(script_file):
+    with open(script_file,'r',encoding='utf-8') as f:
+        contents=f.readlines()
+        for line in contents:
+            if line.startswith('# Source repo') :
+                github_url = ":".join(line.split(':')[1:]).strip()
+                return github_url
+    return "NUrl"
+
+def check_repo_activeness(package_url):
+    owner, repo = package_url.replace('.git','').split('/')[-2:]
+    active_response=requests.get(GITHUB_PACKAGE_INFO_API.format(owner,repo,'branches','master'))
```

Comment:
In case the default branch for any package is not `master `but something like `main`, will the code function smoothly?